packages/docsearch-react/src/utils/buildInsightsClickParams.ts

```diff
@@ -0,0 +1,22 @@
+import type { AlgoliaInsightsHit } from '@algolia/autocomplete-core';
+import type {
+  InsightsParamsWithItems,
+  ClickedObjectIDsAfterSearchParams,
+} from '@algolia/autocomplete-plugin-algolia-insights';
+
+import type { InternalDocSearchHit } from '../types';
+
+export function buildInsightsClickParams(
+  item: InternalDocSearchHit,
+  itemPosition: number
+): InsightsParamsWithItems<ClickedObjectIDsAfterSearchParams> {
+  const insightsItem = item as AlgoliaInsightsHit;
```

Comment:
Could we run into issues with this assertion?

Reply:
I don't think so, if we have an `insights` plugin active the needed attributes are added to the `item`. All `AlgoliaInsightsHit` type adds is better typings for the `__autocomplete_*` attributes.